Feat/caching

Uacias opened the pull request:
* observed wallets are live
* names of wallets are live
* balances of each token per wallet are live

Commit:
in progress

Commit:
Fix wallet.surql, add that file to database-provision.sh

Fix isExecutable type from boolean to bool.
Add the file to script so it is set up on init.

Commit:
Add wallet-balance.surql

Commit:
in progress

Commit:
Currently partially working solution for caching, got:

- Cache for all observed wallets of user (little fix needed because of edge)
- Cache names of observed wallets
- Cache balances of tokens per wallet

Commit:
Fix quantity

Quantity of each token now is converted from Wei.

Commit:
rm crossenv

Commit:
small fixes

Commit:
fix live queries

Commit:
Merge branch 'dev' into feat/caching

Commit:
https, http2 for express

Commit:
tmp fix configs, gotta work on refresh watchaccount

neotheprogramist commented on fixtures/wallet-balance.surql:
```diff
@@ -0,0 +1,6 @@
+DEFINE TABLE wallet_balance SCHEMALESS;
```
while defining don't we prefer `SCHEMAFULL` instead of `SCHEMALESS`?

Uacias replied:
on point, modified all `.surql` files

neotheprogramist commented on src/components/Tokens/TokenRowWallets.tsx:
```diff
@@ -22,8 +36,122 @@ type TokenRowWalletsProps = {
+  await db.query(
+    `DELETE FROM queryuuids WHERE queryUuid = '${latestTokenPriceQueryUuid[0]}';`,
+  );
+  return;
```
unnecessary `return` statement

Uacias replied:
mb, fixed

neotheprogramist commented on src/components/Tokens/TokenRowWallets.tsx:
```diff
@@ -34,6 +162,55 @@ export const TokenRowWallets = component$<TokenRowWalletsProps>(
+      }
+      const data = await tokenRowWalletsInfoStream(walletId, symbol);
+
+      cleanup(async () => {
```
I would always move registration of `cleanup` to the very top of the function (this may be standard to increase code readability)

Uacias replied:
on point, fixed

neotheprogramist commented on src/components/Wallets/Details/SelectedWalletDetails.tsx:
```diff
@@ -7,51 +14,72 @@ import { TokenRowWallets } from "~/components/Tokens/TokenRowWallets";
   }) => {
-    if (!selectedWallet.value) return <></>;
-    let shortAddress = selectedWallet.value.wallet.address;
+    const selectedWalletFromContext = useContext(SelectedWalletDetailsContext);
```
I would rename `selectedWalletFromContext` to `selectedWalletDetails`

Uacias replied:
fixed

Commit:
CR fixes

Rename selectedWalletFromContext to selectedWalletDetails in SelectedWalletDetails component for readability.
Move cleanup to the top of useVisibleTasks.
Rm unnecessary return statement from TokenRowWallets component.
Modify surql files table definitions to SCHEMAFULL from SCHEMALESS.

Commit:
fmt

Commit:
Total portfolio value

Add total portfolio value fn, to calculate latest balance of every observed wallet per user.
Add get exact tick dates function, do get dates for selected period eg. day, week, month, year.

Commit:
Add current moment to getPortfolioDatesForSelectedPeriod so we have fully covered ticks scenarios.

Commit:
small changes

Uacias committed:
Fast working dashboard.

Add fetching all data for charts, time reduced to ~1s, can be optimized by creating queries then batch query to db.
Fix dashboard charts bugs.
Add deleting wallet.

Uacias committed:
lint fmt

Uacias committed:
code cleanup

Gawc1uuu committed:
optimalization of  queries
